presto-bytecode/src/main/java/com/facebook/presto/bytecode/control/SwitchStatement.java

```diff
@@ -41,23 +41,31 @@ public static SwitchBuilder switchBuilder()
         return new SwitchBuilder();
     }
 
+    public static SwitchBuilder switchBuilder(boolean skipEndLabel)
+    {
+        return new SwitchBuilder(skipEndLabel);
+    }
+
     private final LabelNode endLabel = new LabelNode("switchEnd");
     private final LabelNode defaultLabel = new LabelNode("switchDefault");
     private final String comment;
     private final BytecodeExpression expression;
     private final SortedSet<CaseStatement> cases;
     private final BytecodeNode defaultBody;
+    private final boolean skipEndLabel;
 
     private SwitchStatement(
             String comment,
             BytecodeExpression expression,
             Iterable<CaseStatement> cases,
-            BytecodeNode defaultBody)
+            BytecodeNode defaultBody,
+            boolean skipEndLabel)
```

Comment:
hmmmm under what circumstances will we skip the end label?

Reply:
Was using switch statement to generate goto label at the end of cse evaluation. Because each case is a GOTO there's no need for another GOTO. @kaikalur was complaining about the generated bytecode looks strange so I added this. Anyways this change is removed as I'm abandoning the switch + GOTO approach and opted for cse evaluation functions instead.